src/main/java/com/blackducksoftware/integration/hub/validator/HubServerConfigValidator.java

```diff
@@ -129,11 +113,21 @@ public ValidationResults assertProxyValid() {
         return results;
     }
 
-    public ValidationResults assertCredentialsValid() {
+    public ValidationResults assertCredentialsOrApiKeyValid() {
         final CredentialsValidator credentialsBuilder = new CredentialsValidator();
         credentialsBuilder.setUsername(username);
         credentialsBuilder.setPassword(password);
-        return credentialsBuilder.assertValid();
+        final ValidationResults credentialsResults = credentialsBuilder.assertValid();
+        if (credentialsResults.hasErrors()) {
+            // you can specify either username/password OR apiKey
+            if (StringUtils.isNotBlank(apiKey)) {
```

Comment:
Should this supersede username/password? If so we can switch the validation order.

Reply:
If we specify the check for API key it would potentially operate a faster if we are using the API key.  We wouldn't spend cycles validating the invalid credentials.  For that reason I would like the if (StringUtils.isNotBlank(apiKey) {

} else {
//credentials check....
//check for errors and add missing api key if errors found
}